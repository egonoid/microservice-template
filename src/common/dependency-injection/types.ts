export const PingService = Symbol.for('PingService');
export const EventService = Symbol.for('EventService');
