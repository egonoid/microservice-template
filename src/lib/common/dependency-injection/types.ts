export const PingService = Symbol.for('PingService');
