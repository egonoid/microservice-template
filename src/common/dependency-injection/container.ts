import { Container } from 'inversify';
import * as Types from './types';
import { IPingService } from '../../application/services/interfaces/ping.service';
import PingService from '../../application/services/ping.service';

const container = new Container({ skipBaseClassChecks: true });

container.bind<IPingService>(Types.PingService).to(PingService);

export default container;
