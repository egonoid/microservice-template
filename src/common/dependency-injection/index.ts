import * as DITypes from './types';

export { default as DIContainer } from './container';
export { DITypes };
