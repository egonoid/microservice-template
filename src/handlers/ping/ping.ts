import { APIGatewayProxyEvent, APIGatewayProxyHandler } from 'aws-lambda';
import 'source-map-support/register';

export const ping: APIGatewayProxyHandler = async (
  event: APIGatewayProxyEvent
) => {
  const body = event.queryStringParameters;
  const message = body && body.message ? body.message : 'pong';

  return {
    statusCode: 200,
    body: JSON.stringify(
      {
        message,
      },
      null,
      2
    ),
  };
};

export default ping;
