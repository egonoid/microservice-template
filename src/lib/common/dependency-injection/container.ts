import { Container } from 'inversify';
import * as Types from './types';
import { IPingService } from '../../application/interfaces/ping.service';
import PingService from '../../application/ping.service';

const container = new Container({ skipBaseClassChecks: true });

container.bind<IPingService>(Types.PingService).to(PingService);

export default container;
