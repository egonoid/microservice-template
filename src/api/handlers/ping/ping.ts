import 'reflect-metadata';
import { APIGatewayProxyEvent, APIGatewayProxyHandler } from 'aws-lambda';
import 'source-map-support/register';
import { IPingService } from '../../../application/services/interfaces/ping.service';
import * as Types from '../../../common/dependency-injection/types';
import container from '../../../common/dependency-injection/container';
import middy from 'middy';
import eventLogger from '../../middlewares/eventLogger.middleware';

export const ping: APIGatewayProxyHandler = async (
  event: APIGatewayProxyEvent
) => {
  const body = event.queryStringParameters;
  const message = body && body.message;
  const pingService = container.get<IPingService>(Types.PingService);
  const result = pingService.ping(message);

  return {
    statusCode: 200,
    body: JSON.stringify({
      message: result,
    }),
  };
};

export default middy(ping).use(eventLogger());
