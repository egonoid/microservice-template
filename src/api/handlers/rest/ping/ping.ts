import 'reflect-metadata';
import {
  APIGatewayProxyEvent,
  APIGatewayProxyHandler,
  APIGatewayProxyResult,
} from 'aws-lambda';
import 'source-map-support/register';
import { IPingService } from '@application/services/interfaces/ping.service';
import * as Types from '@common/dependency-injection/types';
import container from '@common/dependency-injection/container';
import middy from 'middy';
import { cors } from 'middy/middlewares';
import { eventLogger } from '@egonoid/api-middlewares';
import { v4 as uuid } from 'uuid';

export const ping: APIGatewayProxyHandler = async (
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> => {
  const body = event.queryStringParameters;
  const message = body?.message;
  const pingService = container.get<IPingService>(Types.PingService);
  const result = pingService.ping(message);

  return {
    statusCode: 200,
    body: JSON.stringify({
      message: result,
      uuid: uuid(),
    }),
  };
};

export default middy(ping).use(eventLogger()).use(cors());
