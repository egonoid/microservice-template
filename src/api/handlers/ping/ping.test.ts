import ping from './ping';
import { Context } from 'aws-lambda';
import createEvent from '@serverless/event-mocks';
import { IPingService } from '../../../application/services/interfaces/ping.service';
import * as Types from '../../../common/dependency-injection/types';
import container from '../../../common/dependency-injection/container';
import PingService from '../../../application/services/ping.service';

describe('handler', () => {
  beforeAll(() => {
    const PingServiceMock = jest.fn<PingService, any>(() => ({
      ping: jest.fn().mockReturnValue('pong'),
    }));
    container
      .rebind<IPingService>(Types.PingService)
      .toConstantValue(new PingServiceMock());
  });

  it('should run without error', async () => {
    // Arrange
    const data = {};
    const event = createEvent('aws:apiGateway', data as any);
    const context: Context = {} as Context;
    const callback = null;

    // Act
    const result = await ping(event, context, callback);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
