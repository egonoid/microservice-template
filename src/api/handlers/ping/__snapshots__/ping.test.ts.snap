// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler should return the "pong" when no message is passed in 1`] = `
Object {
  "body": "{
  \\"message\\": \\"pong\\"
}",
  "statusCode": 200,
}
`;

exports[`handler should return the passed in message 1`] = `
Object {
  "body": "{
  \\"message\\": \\"test\\"
}",
  "statusCode": 200,
}
`;
